lib/kitchen_sink/algorithms.ex

```diff
@@ -67,58 +67,112 @@ defmodule KitchenSink.Algorithms do
       iex> Float.round(result, 6)
       10.311285
   """
+  def binary_search(start, finish, fit, target, strategy \\ :midpoint)
   @spec binary_search(
     binary_search_position, binary_search_position, binary_search_fit_func, any, binary_search_strategy
   ) :: binary_search_result
+  def binary_search(
+    range_start_list,
```

Comment:
Can we make this a List of touples instead of 2 lists? Seems a lot easier to reason about.

Reply:
I had a look at doing it, but it would make `defp do_binary_search(position, position, fit, target, _), do: ok_or_not_found(position, fit, target)` untenable. I'd have to rewrite this check. I can do it, but is it worth the time/effort?